components/ContactForm.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { useState, useEffect } from 'react';
+import Select from 'react-select';
+import styles from '../styles/components/ContactForm.module.scss';
+
+interface TextInputProps {
+  label: string;
+  type: string;
+  placeholder: string;
+  value: string;
+  long?: boolean;
+  mandatory?: boolean;
+}
+
+interface DropdownInputProps {
+  label: string;
+  options: { value: string; label: string }[];
+  value: { value: string; label: string } | null;
+  mandatory?: boolean;
+}
+
+interface GroupProps {
+  inputs: (TextInputProps | DropdownInputProps)[];
+}
+
+type FormItem = TextInputProps | DropdownInputProps | GroupProps;
+
+type SubmissionState = { success: true } | { success: false; message: string };
+
+interface ContactFormProps {
+  title: string;
+  description: string;
+  formItems: FormItem[];
+  onSubmit: (formData: Record<string, string>) => Promise<void | Response>;
+  formData: Record<string, string>;
+  onInputChange: (label: string, value: string) => void;
+}
+
+const ContactForm: React.FC<ContactFormProps> = ({
+  title,
+  description,
+  formItems,
+  onSubmit,
+  formData,
+  onInputChange,
+}) => {
+  const [touchedFields, setTouchedFields] = useState<Record<string, boolean>>(
+    {}
+  );
+  const [canSubmit, setCanSubmit] = useState<boolean>(false);
+  const [submissionState, setSubmissionState] =
+    useState<SubmissionState | null>(null);
+
+  useEffect(() => {
+    validateForm();
+  }, [formData, touchedFields]);
+
+  const handleBlur = (label: string) => {
+    setTouchedFields((prevTouched) => ({ ...prevTouched, [label]: true }));
+  };
+
+  const validateEmail = (email: string): boolean => {
+    // This regex comes from https://stackoverflow.com/a/46181
+    // see throrough evaluation of the regex here: https://jsfiddle.net/ghvj4gy9/
+    const emailRegex =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return emailRegex.test(email);
+  };
+
+  const validateForm = () => {
+    let isValid = true;
+
+    const validateItems = (items: any[]) => {
+      items.forEach((item) => {
+        if ('inputs' in item) {
```

Comment:
could we make an enum like 
```
enum ElementType {
  Group,
  Item
}
``` 
and then could you build a tree like 
```
interface TextItem {
  nodeType: Item;
  data: here // ...
}
interface GroupItem {
  nodeType: Group; // note you cannot extend something with the nodeType ElementType because it breaks type narrowing.
  date: here// ...
}
type ContactItem = TextItem | GroupItem; 
``` 
this would allow this to be a switch statement and ensure a lot more typesafety.